drivers/sensor/ti/tmp1075/tmp1075_trigger.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 Arrow Electronics.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+#include "tmp1075.h"
+
+LOG_MODULE_DECLARE(TMP1075, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * @brief GPIO alert line interrupt calback
+ * @param gpio - not used
+ * @param cb - callback structure for interrupt handler
+ * @param pins - not used
+ */
+void tmp1075_trigger_handle_alert(const struct device *gpio, struct gpio_callback *cb,
```

Comment:
i'm not certain it's not overcomplicating things that these two functions are here and not in tmp1075.c, it's a simple device driver in the end, there are no #ifdefs in .h to force this trigger support to be necessary